src/main/groovy/edu/oregonstate/mist/osuevents/db/LocalistDAO.groovy

```diff
@@ -0,0 +1,455 @@
+
+        if (response.statusLine.statusCode == HttpStatus.SC_NOT_FOUND) {
+            EntityUtils.consumeQuietly(response.entity)
+            null
+        } else {
+            String responseEntity = EntityUtils.toString(response.entity)
+
+            SingleCommunityResponse community = objectMapper.readValue(
+                    responseEntity, SingleCommunityResponse)
+
+            Campus.fromCommunity(community.community)
+        }
+    }
+
+    /**
+     * Get multiple locations as well as pagination information.
+     * @param pageNumber
+     * @param pageSize
+     * @return
+     */
+    PaginatedLocations getLocations(Integer pageNumber, Integer pageSize) {
+        HttpResponse response = getResponse(placesEndpoint, pageNumber, pageSize)
+
+        String responseEntity = EntityUtils.toString(response.entity)
+
+        MultiplePlacesResponse places = objectMapper.readValue(
+                responseEntity, MultiplePlacesResponse)
+
+        new PaginatedLocations(
+                locations: places.places.collect {
+                    Location.fromPlace(it.place)
+                },
+                paginationObject: places.paginationObject
+        )
+    }
+
+    /**
+     * Get single location by ID.
+     * @param id
+     * @return
+     */
+    Location getlocationByID(String id) {
+        HttpResponse response = getResponse("${placesEndpoint}/${id}")
+
+        if (response.statusLine.statusCode == HttpStatus.SC_NOT_FOUND) {
+            EntityUtils.consumeQuietly(response.entity)
+            null
+        } else {
+            String responseEntity = EntityUtils.toString(response.entity)
+
+            SinglePlaceResponse place = objectMapper.readValue(
+                    responseEntity, SinglePlaceResponse)
+
+            Location.fromPlace(place.place)
+        }
+    }
+
+    /**
+     * Get multiple departments as well as pagination information.
+     * @param pageNumber
+     * @param pageSize
+     * @return
+     */
+    PagniatedDepartments getDepartments(Integer pageNumber, Integer pageSize) {
+        HttpResponse response = getResponse(departmentsEndpoint, pageNumber, pageSize)
+
+        String responseEntity = EntityUtils.toString(response.entity)
+
+        MultipleDepartmentsResponse departments = objectMapper.readValue(
+                responseEntity, MultipleDepartmentsResponse)
+
+        new PagniatedDepartments(
+                departments: departments.departments.collect {
+                    edu.oregonstate.mist.osuevents.core.Department.fromLocalistDepartment(
+                            it.department)
+                },
+                paginationObject: departments.paginationObject
+        )
+    }
+
+    /**
+     * Get multiple departments as well as pagination information.
+     * @param pageNumber
+     * @param pageSize
+     * @return
+     */
+    edu.oregonstate.mist.osuevents.core.Department getDepartmentByID(String id) {
```

Comment:
Is there any reason that you want to import the class like this way? Why not import it at the beginning of this file?

Reply:
This file also contains a class called `Department`, which represents a department as it comes from localist. This method returns the wrapper API's version of a department, which is different. I could probably avoid explicitly typing an object by making the name something different, but I didn't want to make the name different just to make the code cleaner.